src/utils.ts

```diff
@@ -114,7 +113,12 @@ export function executeWithFreshModules<T>(f: () => T): T {
     jest.isolateModules(() => {
       // Ensure every page gets the same patched 'next/document' instance
       // imported before tests by "src/testHelpers.ts"
-      jest.mock('next/document', () => jest.requireActual('next/document'));
+      jest.doMock('react', () => jest.requireActual('react'));
```

Comment:
As this list is increasing, I think we could extract it to some list to always be sure they are in sync with the `stealthyRequire`.

```js
const nonIsolatedModules = ['react, 'react-dom']


nonIsolatedModules.forEach((nonIsolatedModule) => {
   jest.doMock(nonIsolatedModule, () => jest.requireActual(nonIsolatedModule));
})

nonIsolatedModules.forEach((nonIsolatedModule) => {
   require(nonIsolatedModule)
})
 ```

Reply:
I definitely overdid with these non isolated dependencies. `react` and `next/document` were enough. The issue was `jest.requireActual` expecting the module being imported beforehand.